### DISCUSSION TITLE
Jalali Calendar for PrestaShop 1.7

### DISCUSSION THREAD

**Discussion by davoud-najjari** [PrestaShop/PrestaShop#40186] [Ideas]
Please add Jalali Calendar for Prestashop 1.7 for RTL languages such as Persian/Arabic

Thank you

**Comment by khouloudbelguith** [comment]
Hi @davoud-najjari,
After installing the Arabic package from the localization page, in the product detail page, quantity tab, for the Availability date we can find a calendar for this RTL language.
![image](https://user-images.githubusercontent.com/16067358/49284752-d1aed300-f495-11e8-9119-70a8b9ecce3c.png)
Thanks to check & feedback.

**Comment by davoud-najjari** [comment]
Hi khouloudbelguith, 
Thank you for your reply, but I Persian language 'Jalali Date' is completely different 
please have look at below resources for more info regarding this issue 
1- https://github.com/babakhani/PersianDate
2- https://www.drupal.org/project/calendar_systems

for instance 👍  after loading the Persian language inside the localization tab in Prestashop, automatically Perian Date (Jalali Date) will be added to RTL languages 

Thank you 
awaiting to your respond

**Comment by khouloudbelguith** [comment]
@davoud-najjari, thanks for this clarifications.
After importing the Persian language,
I have this calendar 
![image](https://user-images.githubusercontent.com/16067358/49286683-3ae51500-f49b-11e8-876e-0b9bfc6cf4ac.png)
Thanks to check & feedback.

**Comment by davoud-najjari** [comment]
Hi, thank you
You know as you see in the above picture, Year still shows in English, not in Persian (Jalali Date).
I have a picture of Persian Date please have look at below pic.
I think Persian date must be implemented into Prestashop with specific PHP class.  for more information please have look at : 
![datepicker_persian](https://user-images.githubusercontent.com/23167257/49325027-565e2780-f550-11e8-8437-a81f67f602b1.JPG)

Reference: 
http://babakhani.github.io/PersianWebToolkit/doc/datepicker/example/

Thank you
awaiting to your respond

**Comment by khouloudbelguith** [comment]
Hi @davoud-najjari,

Could you please provide the exact version PrestaShop did you installed & which Language/country did you used?
When the installation is complete, did you imported any lanaguage?
We need more details to understand how we can reproduce your issue.
Thanks!

**Comment by davoud-najjari** [comment]
Hi, 
I have installed Prestashop Version 1.7.4.4 latest stable release. 
Step1:
install PrestaShop on the English language 
Step2:
Add language (Persian), Everything is fine. (RTL is correct, Font is fine,...)
Step3:
But PersianDate is not showing correctly as it should be.  for example, Jalali date is completely different from Gregorian Calendar systems.
for clarification, I have gathered some info 👍 
1-We need Supports Iranian/Persian, Arabic/Hijri/Islamic, Thai, Gregorian Calendar systems.
2- We need Supports Multi-language websites via the locale module. Each language can have a different calendar system
3- Please Provided PersianDate class extends Standard PHP DateTime class and has it's all capabilities.
4- for better understanding I write below code 
@@@I use php intl extension for this code. @@@

![image](https://user-images.githubusercontent.com/23167257/49426355-92de8d00-f7b5-11e8-974a-f88df7e15a1b.png)

 Kind regards
 awaiting to your respond

**Comment by khouloudbelguith** [comment]
Hi @davoud-najjari, thanks for this clarifications.
@marionf, @colinegin what do you think, is that possible to support Iranian/Persian, Arabic/Hijri/Islamic, Thai, Gregorian Calendar systems?
![image](https://user-images.githubusercontent.com/16067358/49430030-3e81e000-f7aa-11e8-840a-699262138822.png)
Thanks!

**Comment by MatShir** [comment]
@davoud-najjari Thanks you for your feedback, other people have made the same feed around the Jalali calendar. It will be considered by the team.

**Comment by dn1353** [comment]
> @davoud-najjari Thanks you for your feedback, other people have made the same feed around the Jalali calendar. We will be considered by the team.

@MatShir 
Thank you, Sir,
I found the free and Open Source JavaScript Calendar. (https://fullcalendar.io) that support All localized Dates such as Persian. 
I wish we cloud integrate this library to Prestashop 1.7.7.x
![fullcalendar](https://user-images.githubusercontent.com/62265204/84228555-95cde280-aafc-11ea-8dae-40d3f7b3c59b.JPG)

**Comment by davoud-najjari** [comment]
Hi, thank you for reply <
but as you know Persian date works with the Jalali calendar that doesn't
exist in the Arabic language. the name of the months in the Persian
language must be translated to Persian as well.
for better understanding please have look at below links :
1- https://www.drupal.org/project/calendar_systems
2- http://babakhani.github.io/PersianWebToolkit/doc/datepicker/example/
thank you

**Comment by dn1353** [comment]
Hi, 
any news regarding to implementing fullcalendar.io in Prestashop 1.7.7.x?

**Comment by mparvazi** [comment]
PS use this plugin for calendar: https://github.com/moment/moment/

There is a "Solar Hijri calendar" plugin for moment.js: https://github.com/jalaali/moment-jalaali
